<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Additional topics | Polygenic indices</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Additional topics | Polygenic indices" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Additional topics | Polygenic indices" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="" />


<meta name="date" content="2023-11-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="in-practice.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Polygenic Scores</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="in-practice.html"><a href="in-practice.html"><i class="fa fa-check"></i><b>2</b> In practice</a>
<ul>
<li class="chapter" data-level="2.1" data-path="in-practice.html"><a href="in-practice.html#computing-environment"><i class="fa fa-check"></i><b>2.1</b> Computing environment</a></li>
<li class="chapter" data-level="2.2" data-path="in-practice.html"><a href="in-practice.html#working-with-prsice"><i class="fa fa-check"></i><b>2.2</b> Working with PRSICE</a>
<ul>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-1-prepare-genotypes"><i class="fa fa-check"></i>Step 1: Prepare genotypes</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-2-prepare-summary-statistics"><i class="fa fa-check"></i>Step 2: Prepare summary statistics</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3-create-the-score"><i class="fa fa-check"></i>Step 3: Create the score</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-4-evaluate-the-score"><i class="fa fa-check"></i>Step 4: Evaluate the score</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="in-practice.html"><a href="in-practice.html#working-with-bigsnpr-and-ldpred"><i class="fa fa-check"></i><b>2.3</b> Working with bigsnpr and LDPred</a>
<ul>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-1-prepare-genotypes-1"><i class="fa fa-check"></i>Step 1: Prepare genotypes</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-2-prepare-summary-statistics-1"><i class="fa fa-check"></i>Step 2: Prepare summary statistics</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3-match-summary-statistics-to-genotypes"><i class="fa fa-check"></i>Step 3: Match summary statistics to genotypes</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-4-calculate-ld-information"><i class="fa fa-check"></i>Step 4: Calculate LD information</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-5-calculate-adjusted-weights-using-ldpred-inf"><i class="fa fa-check"></i>Step 5: Calculate adjusted weights using LDPred Inf</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-6-calculate-pgs-in-the-analysis-sample"><i class="fa fa-check"></i>Step 6: Calculate PGS in the analysis sample</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-7-evaluating-the-score"><i class="fa fa-check"></i>Step 7: Evaluating the score</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="additional-topics.html"><a href="additional-topics.html"><i class="fa fa-check"></i><b>3</b> Additional topics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="additional-topics.html"><a href="additional-topics.html#splitting-your-genotype-dataset"><i class="fa fa-check"></i><b>3.1</b> Splitting your genotype dataset</a></li>
<li class="chapter" data-level="3.2" data-path="additional-topics.html"><a href="additional-topics.html#calling-plink-and-cli-utilities-from-r"><i class="fa fa-check"></i><b>3.2</b> Calling PLINK and CLI utilities from R</a></li>
<li class="chapter" data-level="3.3" data-path="additional-topics.html"><a href="additional-topics.html#reading-bgen-files-from-bigsnpr"><i class="fa fa-check"></i><b>3.3</b> Reading BGEN files from bigsnpr</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Polygenic indices</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="additional-topics" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Additional topics<a href="additional-topics.html#additional-topics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="splitting-your-genotype-dataset" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Splitting your genotype dataset<a href="additional-topics.html#splitting-your-genotype-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Say you are using the UKB and want to split your dataset so you can use some of it for your analysis and another part for tuning you polygenic score. How do you do that?</p>
<p>You know the .fam file contains information on the individuals, so let’s load that up in R</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="additional-topics.html#cb72-1" tabindex="-1"></a>fam <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&#39;genotypes.fam&#39;</span>);</span>
<span id="cb72-2"><a href="additional-topics.html#cb72-2" tabindex="-1"></a><span class="fu">head</span>(fam);</span></code></pre></div>
<p>Then use whatever splitting procedure you fancy. Say we want <span class="math inline">\(k\)</span> equal sized chunks (if possible):</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="additional-topics.html#cb73-1" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="dv">5</span>;</span>
<span id="cb73-2"><a href="additional-topics.html#cb73-2" tabindex="-1"></a><span class="fu">split</span>(fam<span class="sc">$</span>IID, <span class="fu">ceiling</span>(<span class="fu">seq_along</span>(<span class="fu">nrow</span>(fam))<span class="sc">/</span><span class="dv">20</span>));</span></code></pre></div>
<p>or perhaps chunks of size <span class="math inline">\(n\)</span></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="additional-topics.html#cb74-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">5</span>;</span>
<span id="cb74-2"><a href="additional-topics.html#cb74-2" tabindex="-1"></a>XXXXXXX</span></code></pre></div>
<p>etc. We can then save these out to a text file and use plink to extract these individuals from the genotypes dataset and create new genotype files for these subsamples.</p>
</div>
<div id="calling-plink-and-cli-utilities-from-r" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Calling PLINK and CLI utilities from R<a href="additional-topics.html#calling-plink-and-cli-utilities-from-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you are like me, and don’t want to leave R, then you also call this from inside R. One advantage of this is that you can use all the usual R functions to set up your call to plink. For example, to run the plink command from earlier, we can execute the following line in R:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="additional-topics.html#cb75-1" tabindex="-1"></a><span class="fu">system</span>(glue<span class="sc">::</span><span class="fu">glue</span>(</span>
<span id="cb75-2"><a href="additional-topics.html#cb75-2" tabindex="-1"></a>  <span class="st">&#39;plink --bfile geno --extract snplist.txt --make-bed --out filtered_geno&#39;</span></span>
<span id="cb75-3"><a href="additional-topics.html#cb75-3" tabindex="-1"></a>));</span></code></pre></div>
<p>This is a bit contrived for such a simple case, but say you want to execute the same command for each chromosome? And you also want to create the SNP list in R first.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="additional-topics.html#cb76-1" tabindex="-1"></a><span class="co"># write a list of snps to a file</span></span>
<span id="cb76-2"><a href="additional-topics.html#cb76-2" tabindex="-1"></a>snpfile <span class="ot">&lt;-</span> <span class="st">&#39;snplist.txt&#39;</span>;</span>
<span id="cb76-3"><a href="additional-topics.html#cb76-3" tabindex="-1"></a></span>
<span id="cb76-4"><a href="additional-topics.html#cb76-4" tabindex="-1"></a><span class="fu">write.table</span>(</span>
<span id="cb76-5"><a href="additional-topics.html#cb76-5" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">ids =</span> <span class="fu">c</span>(<span class="st">&#39;rs100000&#39;</span>, <span class="st">&#39;rs1000001&#39;</span>, <span class="st">&#39;rs1000002&#39;</span>)),</span>
<span id="cb76-6"><a href="additional-topics.html#cb76-6" tabindex="-1"></a>  <span class="at">file      =</span> snpfile,</span>
<span id="cb76-7"><a href="additional-topics.html#cb76-7" tabindex="-1"></a>  <span class="at">quote     =</span> <span class="cn">FALSE</span>,</span>
<span id="cb76-8"><a href="additional-topics.html#cb76-8" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb76-9"><a href="additional-topics.html#cb76-9" tabindex="-1"></a>  <span class="at">col.names =</span> <span class="cn">FALSE</span></span>
<span id="cb76-10"><a href="additional-topics.html#cb76-10" tabindex="-1"></a>);</span>
<span id="cb76-11"><a href="additional-topics.html#cb76-11" tabindex="-1"></a></span>
<span id="cb76-12"><a href="additional-topics.html#cb76-12" tabindex="-1"></a><span class="cf">for</span> (chr <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>) {</span>
<span id="cb76-13"><a href="additional-topics.html#cb76-13" tabindex="-1"></a>  <span class="fu">system</span>(</span>
<span id="cb76-14"><a href="additional-topics.html#cb76-14" tabindex="-1"></a>    <span class="st">&#39;plink --bfile genoChr{chr} --extract {snpfile} --make-bed --out filterChr{chr}&#39;</span></span>
<span id="cb76-15"><a href="additional-topics.html#cb76-15" tabindex="-1"></a>  );</span>
<span id="cb76-16"><a href="additional-topics.html#cb76-16" tabindex="-1"></a>}</span></code></pre></div>
<p>This will generate and execite the following command line calls to plink:</p>
<pre class="default-output-style"><code>plink --bfile genoChr1 --extract snplist.txt --make-bed --out filterChr1 
plink --bfile genoChr2 --extract snplist.txt --make-bed --out filterChr2 
plink --bfile genoChr3 --extract snplist.txt --make-bed --out filterChr3 
plink --bfile genoChr4 --extract snplist.txt --make-bed --out filterChr4 
plink --bfile genoChr5 --extract snplist.txt --make-bed --out filterChr5 
plink --bfile genoChr6 --extract snplist.txt --make-bed --out filterChr6 
plink --bfile genoChr7 --extract snplist.txt --make-bed --out filterChr7 
plink --bfile genoChr8 --extract snplist.txt --make-bed --out filterChr8 
plink --bfile genoChr9 --extract snplist.txt --make-bed --out filterChr9 
plink --bfile genoChr10 --extract snplist.txt --make-bed --out filterChr10 
plink --bfile genoChr11 --extract snplist.txt --make-bed --out filterChr11 
plink --bfile genoChr12 --extract snplist.txt --make-bed --out filterChr12 
plink --bfile genoChr13 --extract snplist.txt --make-bed --out filterChr13 
plink --bfile genoChr14 --extract snplist.txt --make-bed --out filterChr14 
plink --bfile genoChr15 --extract snplist.txt --make-bed --out filterChr15 
plink --bfile genoChr16 --extract snplist.txt --make-bed --out filterChr16 
plink --bfile genoChr17 --extract snplist.txt --make-bed --out filterChr17 
plink --bfile genoChr18 --extract snplist.txt --make-bed --out filterChr18 
plink --bfile genoChr19 --extract snplist.txt --make-bed --out filterChr19 
plink --bfile genoChr20 --extract snplist.txt --make-bed --out filterChr20 
plink --bfile genoChr21 --extract snplist.txt --make-bed --out filterChr21 
plink --bfile genoChr22 --extract snplist.txt --make-bed --out filterChr22 </code></pre>
<p>Saved us a lot of typing or error prone copy-pasting! You can do this for any system call, so it’s also applicable when we use PRSICE.</p>
</div>
<div id="reading-bgen-files-from-bigsnpr" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Reading BGEN files from bigsnpr<a href="additional-topics.html#reading-bgen-files-from-bigsnpr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="additional-topics.html#cb78-1" tabindex="-1"></a>bgenfiles_paths    <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(ukb_bgen_pattern, <span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>);</span>
<span id="cb78-2"><a href="additional-topics.html#cb78-2" tabindex="-1"></a>bgifiles_directory <span class="ot">&lt;-</span> <span class="fu">dirname</span>(bgenfiles_paths);</span>
<span id="cb78-3"><a href="additional-topics.html#cb78-3" tabindex="-1"></a>bgifiles_directory <span class="ot">&lt;-</span> <span class="fu">rep</span>(</span>
<span id="cb78-4"><a href="additional-topics.html#cb78-4" tabindex="-1"></a>  ukb_bgen_bgi_dir,</span>
<span id="cb78-5"><a href="additional-topics.html#cb78-5" tabindex="-1"></a>  <span class="fu">length</span>(bgenfiles_paths)</span>
<span id="cb78-6"><a href="additional-topics.html#cb78-6" tabindex="-1"></a>);</span>
<span id="cb78-7"><a href="additional-topics.html#cb78-7" tabindex="-1"></a></span>
<span id="cb78-8"><a href="additional-topics.html#cb78-8" tabindex="-1"></a>path_to_rds <span class="ot">&lt;-</span> bigsnpr<span class="sc">::</span><span class="fu">snp_readBGEN</span>(</span>
<span id="cb78-9"><a href="additional-topics.html#cb78-9" tabindex="-1"></a>  <span class="at">bgenfiles   =</span> bgenfiles_paths,</span>
<span id="cb78-10"><a href="additional-topics.html#cb78-10" tabindex="-1"></a>  <span class="at">backingfile =</span> path_to_output_rds,</span>
<span id="cb78-11"><a href="additional-topics.html#cb78-11" tabindex="-1"></a>  <span class="co"># list_snp_id = snp_list,</span></span>
<span id="cb78-12"><a href="additional-topics.html#cb78-12" tabindex="-1"></a>  <span class="at">ncores      =</span> number_of_cores,</span>
<span id="cb78-13"><a href="additional-topics.html#cb78-13" tabindex="-1"></a>  <span class="at">bgi_dir     =</span> bgifiles_directory</span>
<span id="cb78-14"><a href="additional-topics.html#cb78-14" tabindex="-1"></a>);</span>
<span id="cb78-15"><a href="additional-topics.html#cb78-15" tabindex="-1"></a></span>
<span id="cb78-16"><a href="additional-topics.html#cb78-16" tabindex="-1"></a><span class="co"># </span><span class="al">HACK</span><span class="co">: We need to manually attach the fam info as we will need it later. We</span></span>
<span id="cb78-17"><a href="additional-topics.html#cb78-17" tabindex="-1"></a><span class="co"># can pick a random fam sample file as they are all the same.</span></span>
<span id="cb78-18"><a href="additional-topics.html#cb78-18" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ukb_sample_information_path)) {</span>
<span id="cb78-19"><a href="additional-topics.html#cb78-19" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(ukb_sample_information_path) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb78-20"><a href="additional-topics.html#cb78-20" tabindex="-1"></a>    ukb_sample_information_path <span class="ot">&lt;-</span> ukb_sample_information_path[<span class="dv">1</span>];</span>
<span id="cb78-21"><a href="additional-topics.html#cb78-21" tabindex="-1"></a>  }</span>
<span id="cb78-22"><a href="additional-topics.html#cb78-22" tabindex="-1"></a></span>
<span id="cb78-23"><a href="additional-topics.html#cb78-23" tabindex="-1"></a>  <span class="fu">l</span>(<span class="st">&#39;loading sample information from %s&#39;</span>, ukb_sample_information_path);</span>
<span id="cb78-24"><a href="additional-topics.html#cb78-24" tabindex="-1"></a></span>
<span id="cb78-25"><a href="additional-topics.html#cb78-25" tabindex="-1"></a>  sample_info <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb78-26"><a href="additional-topics.html#cb78-26" tabindex="-1"></a>    ukb_sample_information_path,</span>
<span id="cb78-27"><a href="additional-topics.html#cb78-27" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-28"><a href="additional-topics.html#cb78-28" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&#39; &#39;</span></span>
<span id="cb78-29"><a href="additional-topics.html#cb78-29" tabindex="-1"></a>  );</span>
<span id="cb78-30"><a href="additional-topics.html#cb78-30" tabindex="-1"></a></span>
<span id="cb78-31"><a href="additional-topics.html#cb78-31" tabindex="-1"></a>  <span class="fu">l</span>(<span class="st">&#39;structure:&#39;</span>);</span>
<span id="cb78-32"><a href="additional-topics.html#cb78-32" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">head</span>(sample_info));</span>
<span id="cb78-33"><a href="additional-topics.html#cb78-33" tabindex="-1"></a></span>
<span id="cb78-34"><a href="additional-topics.html#cb78-34" tabindex="-1"></a>  <span class="co"># first row can be weird for some reason?</span></span>
<span id="cb78-35"><a href="additional-topics.html#cb78-35" tabindex="-1"></a>  first_row <span class="ot">&lt;-</span> sample_info[<span class="dv">1</span>, ];</span>
<span id="cb78-36"><a href="additional-topics.html#cb78-36" tabindex="-1"></a>  <span class="cf">if</span> (</span>
<span id="cb78-37"><a href="additional-topics.html#cb78-37" tabindex="-1"></a>    first_row<span class="sc">$</span>ID_1 <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;&amp;</span> first_row<span class="sc">$</span>ID_2 <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;&amp;</span> first_row<span class="sc">$</span>missing <span class="sc">==</span> <span class="dv">0</span></span>
<span id="cb78-38"><a href="additional-topics.html#cb78-38" tabindex="-1"></a>  ) {</span>
<span id="cb78-39"><a href="additional-topics.html#cb78-39" tabindex="-1"></a>    <span class="fu">l</span>(<span class="st">&#39;discarding first row...&#39;</span>);</span>
<span id="cb78-40"><a href="additional-topics.html#cb78-40" tabindex="-1"></a>    sample_info <span class="ot">&lt;-</span> sample_info[<span class="sc">-</span><span class="dv">1</span>, ];</span>
<span id="cb78-41"><a href="additional-topics.html#cb78-41" tabindex="-1"></a>  }</span>
<span id="cb78-42"><a href="additional-topics.html#cb78-42" tabindex="-1"></a></span>
<span id="cb78-43"><a href="additional-topics.html#cb78-43" tabindex="-1"></a>  <span class="fu">colnames</span>(sample_info) <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb78-44"><a href="additional-topics.html#cb78-44" tabindex="-1"></a>    <span class="st">&#39;family.ID&#39;</span>,</span>
<span id="cb78-45"><a href="additional-topics.html#cb78-45" tabindex="-1"></a>    <span class="st">&#39;sample.ID&#39;</span>,</span>
<span id="cb78-46"><a href="additional-topics.html#cb78-46" tabindex="-1"></a>    <span class="st">&#39;missing&#39;</span>,</span>
<span id="cb78-47"><a href="additional-topics.html#cb78-47" tabindex="-1"></a>    <span class="st">&#39;sex&#39;</span></span>
<span id="cb78-48"><a href="additional-topics.html#cb78-48" tabindex="-1"></a>  );</span>
<span id="cb78-49"><a href="additional-topics.html#cb78-49" tabindex="-1"></a></span>
<span id="cb78-50"><a href="additional-topics.html#cb78-50" tabindex="-1"></a>  <span class="co"># </span><span class="al">HACK</span><span class="co">: add manually to bigSNP object and re-save</span></span>
<span id="cb78-51"><a href="additional-topics.html#cb78-51" tabindex="-1"></a>  bigsnp_object <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(path_to_rds);</span>
<span id="cb78-52"><a href="additional-topics.html#cb78-52" tabindex="-1"></a>  bigsnp_object<span class="sc">$</span>fam <span class="ot">&lt;-</span> sample_info;</span>
<span id="cb78-53"><a href="additional-topics.html#cb78-53" tabindex="-1"></a>  bigsnp_object <span class="ot">&lt;-</span> <span class="fu">saveRDS</span>(bigsnp_object, path_to_rds);</span>
<span id="cb78-54"><a href="additional-topics.html#cb78-54" tabindex="-1"></a></span>
<span id="cb78-55"><a href="additional-topics.html#cb78-55" tabindex="-1"></a>  <span class="co"># make a sanity check</span></span>
<span id="cb78-56"><a href="additional-topics.html#cb78-56" tabindex="-1"></a>  bigsnp_object <span class="ot">&lt;-</span> bigsnpr<span class="sc">::</span><span class="fu">snp_attach</span>(path_to_rds);</span>
<span id="cb78-57"><a href="additional-topics.html#cb78-57" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(bigsnp_object<span class="sc">$</span>genotypes) <span class="sc">!=</span> <span class="fu">nrow</span>(sample_info)) <span class="fu">warning</span>(</span>
<span id="cb78-58"><a href="additional-topics.html#cb78-58" tabindex="-1"></a>    <span class="st">&#39;number of individuals in genotypes did not match number of individuals in sample information.&#39;</span></span>
<span id="cb78-59"><a href="additional-topics.html#cb78-59" tabindex="-1"></a>  );</span>
<span id="cb78-60"><a href="additional-topics.html#cb78-60" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb78-61"><a href="additional-topics.html#cb78-61" tabindex="-1"></a>  <span class="fu">l</span>(<span class="st">&#39;no sample file given, so no sample information is added to genotypes&#39;</span>);</span>
<span id="cb78-62"><a href="additional-topics.html#cb78-62" tabindex="-1"></a>}</span>
<span id="cb78-63"><a href="additional-topics.html#cb78-63" tabindex="-1"></a></span>
<span id="cb78-64"><a href="additional-topics.html#cb78-64" tabindex="-1"></a>path_to_rds;</span></code></pre></div>
<!--

check_genotypes_fam(genotypes);


internal_sample_ids <- bigparallelr::rows_along(genotypes$genotypes);
if (!is.null(sample_ids)) {
  internal_sample_ids <- which(
    genotypes$fam$sample.ID %in% sample_ids
  );
}
-->

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="in-practice.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-additional.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
