<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Background | Polygenic indices</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Background | Polygenic indices" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Background | Polygenic indices" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="" />


<meta name="date" content="2023-11-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="in-practice.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Polygenic Scores</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="background.html"><a href="background.html#genotypes"><i class="fa fa-check"></i><b>2.1</b> Genotypes</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="background.html"><a href="background.html#bed-plink-format"><i class="fa fa-check"></i><b>2.1.1</b> bed – PLINK format</a></li>
<li class="chapter" data-level="2.1.2" data-path="background.html"><a href="background.html#bgen-oxford-format"><i class="fa fa-check"></i><b>2.1.2</b> bgen – Oxford format</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="background.html"><a href="background.html#summary-statistics"><i class="fa fa-check"></i><b>2.2</b> Summary statistics</a></li>
<li class="chapter" data-level="2.3" data-path="background.html"><a href="background.html#datasets"><i class="fa fa-check"></i><b>2.3</b> Datasets</a></li>
<li class="chapter" data-level="2.4" data-path="background.html"><a href="background.html#polygenic-score-methods"><i class="fa fa-check"></i><b>2.4</b> Polygenic score methods</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="background.html"><a href="background.html#naive-method"><i class="fa fa-check"></i><b>2.4.1</b> Naive method</a></li>
<li class="chapter" data-level="2.4.2" data-path="background.html"><a href="background.html#pruningclumping-and-threshold"><i class="fa fa-check"></i><b>2.4.2</b> Pruning/clumping and threshold</a></li>
<li class="chapter" data-level="2.4.3" data-path="background.html"><a href="background.html#ldpred"><i class="fa fa-check"></i><b>2.4.3</b> LDPred</a></li>
<li class="chapter" data-level="2.4.4" data-path="background.html"><a href="background.html#lassosum"><i class="fa fa-check"></i><b>2.4.4</b> LASSOSUM</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="in-practice.html"><a href="in-practice.html"><i class="fa fa-check"></i><b>3</b> In practice</a>
<ul>
<li class="chapter" data-level="3.1" data-path="in-practice.html"><a href="in-practice.html#computing-environment"><i class="fa fa-check"></i><b>3.1</b> Computing environment</a></li>
<li class="chapter" data-level="3.2" data-path="in-practice.html"><a href="in-practice.html#working-with-prsice"><i class="fa fa-check"></i><b>3.2</b> Working with PRSICE</a>
<ul>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-1-preparing-genotypes"><i class="fa fa-check"></i>Step 1: Preparing genotypes</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-2-qcing-summarystatistics"><i class="fa fa-check"></i>Step 2: QC’ing summarystatistics</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3-creating-the-score"><i class="fa fa-check"></i>Step 3: Creating the score</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-4-evaluating-the-score"><i class="fa fa-check"></i>Step 4: Evaluating the score</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="in-practice.html"><a href="in-practice.html#working-with-bigsnpr-and-ldpred"><i class="fa fa-check"></i><b>3.3</b> Working with bigsnpr and LDPred</a>
<ul>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-1-preparing-genotypes-1"><i class="fa fa-check"></i>Step 1: Preparing genotypes</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-2-qcing-summarystatistics-1"><i class="fa fa-check"></i>Step 2: QC’ing summarystatistics</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3-matching-summary-statistics-and-genotypes"><i class="fa fa-check"></i>Step 3: Matching summary statistics and genotypes</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3.2-calculating-the-ld-information"><i class="fa fa-check"></i>Step 3.2: Calculating the LD information</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3-creating-the-score-1"><i class="fa fa-check"></i>Step 3: Creating the score</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3.2-calculating-adjusted-weights-using-ldpred-inf"><i class="fa fa-check"></i>Step 3.2: Calculating adjusted weights using LDPred Inf</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3.3-calculating-adjusted-weights-using-ldpred-and-grid-search"><i class="fa fa-check"></i>Step 3.3: Calculating adjusted weights using LDPred and grid search</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3.3-calculating-adjusted-weights-using-ldpred-auto"><i class="fa fa-check"></i>Step 3.3: Calculating adjusted weights using LDPred auto</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-3.5-calculating-the-pgs-in-the-analysis-sample"><i class="fa fa-check"></i>Step 3.5: Calculating the PGS in the analysis sample</a></li>
<li class="chapter" data-level="" data-path="in-practice.html"><a href="in-practice.html#step-4-evaluating-the-score-1"><i class="fa fa-check"></i>Step 4: Evaluating the score</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="additional-topics.html"><a href="additional-topics.html"><i class="fa fa-check"></i><b>4</b> Additional topics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="additional-topics.html"><a href="additional-topics.html#splitting-your-genotype-dataset"><i class="fa fa-check"></i><b>4.1</b> Splitting your genotype dataset</a></li>
<li class="chapter" data-level="4.2" data-path="additional-topics.html"><a href="additional-topics.html#calling-plink-and-cli-utilities-from-r"><i class="fa fa-check"></i><b>4.2</b> Calling PLINK and CLI utilities from R</a></li>
<li class="chapter" data-level="4.3" data-path="additional-topics.html"><a href="additional-topics.html#reading-bgen-files-from-bigsnpr"><i class="fa fa-check"></i><b>4.3</b> Reading BGEN files from bigsnpr</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Polygenic indices</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="background" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Background<a href="background.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To construct PGS, but this section will give some background on the different PGS methods, and some general things we should consider when QC’ing and working with genotypes and summary statistics.</p>
<p>To construct a polygenic score we need lalala. Below we will discuss first the genotype data, then the summary statistics, and finally the polygenic score methods.</p>
<p>We saw yesterday what a PGS is and how to include it in analyses, but how do you construct them in the first place?</p>
<p>You’ve seen how to QC genotype data.
You’ve seen how to construct a PGS as the sum of SNPs weighted by their GWAS effect size estimates.</p>
<p>Many methods with different levels of complexity:</p>
<ul>
<li>Naïve/basic method – use all effect estimates from GWAS directly.</li>
<li>PT+Clumping (PRSICE) – select specific SNPs to include based on GWAS estimates based on P-value and prune markers based on LD (to try to adjust for LD we don’t adjust effect estimates, instead we throw SNPs out!).</li>
<li>LDPred – select specific SNPs and correct effect estimates for SNP correlations (LD) (we don’t throw markers out to correct for LD but instead adjust the effect sizes!)</li>
<li>LASSOSUM – select specific SNPs and also correct for SNP correlations (LD) (we don’t throw markers out to correct for LD but instead adjust the effect sizes).</li>
<li>The continuous shrinkage PGS method? Can’t remember the name, look up the paper.</li>
</ul>
<p>We will consider questions such as:</p>
<ul>
<li>How do I make the most predictive score?</li>
<li>Which method is the best?</li>
<li>What’s the best choice of parameter XYZ?</li>
<li>As with any prediction problem: Tweak parameters/model, evaluate your model on hold-out data, iterate. Pick the model that scores highest on your evaluation metric.</li>
</ul>
<p>General recipe for a polygenic score:</p>
<ol style="list-style-type: decimal">
<li>We want to construct a PGI for phenotype, say, <span class="math inline">\(y_i\)</span>, where <span class="math inline">\(i\)</span> is individual.</li>
<li>In a discovery sample <span class="math inline">\(\mathcal{D}\)</span>, we run a simple GWAS for <span class="math inline">\(y\)</span>, say:
<span class="math display">\[y_i = \alpha + \beta_{j} SNP_{ij} + \varepsilon_{ij}, \quad i \in \mathcal{D}, j = 1, \ldots, J\]</span>
to get summary statistics <span class="math inline">\({\{}\hat{\beta_j}, se(\hat{\beta_j}){\}}_{j=1}^J\)</span>. Recall that <span class="math inline">\(SNP_{ij}\)</span> denotes an allele count!</li>
<li>In a validation sample <span class="math inline">\(\mathcal{T}\)</span>, we pick a set of SNPs <span class="math inline">\(\mathcal{J}\)</span>, and calculate adjusted effect sizes <span class="math inline">\(\tilde{\beta}_j = f(\hat{\beta}_j), j \in \mathcal{J}\)</span> using some method. In this process we may rely on LD information from a reference sample <span class="math inline">\(\mathcal{R}\)</span>.</li>
<li>Finally, in the analysis sample <span class="math inline">\(\mathcal{A}\)</span>, for each individual <span class="math inline">\(i \in \mathcal{A}\)</span> we construct the score
<span class="math display">\[PGI_i = \sum_{j \in \mathcal{S}} \tilde{\beta}_j SNP_{ij}, \quad i \in \mathcal{A}\]</span></li>
</ol>
<div id="genotypes" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Genotypes<a href="background.html#genotypes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Will assume you have a file with appropriately QC’ed genotypes.</li>
<li>Quality-control: Implement best-practices from GWAS lecture and labs.</li>
<li>Pitfall: Genetic ancestry!</li>
<li>Convenient: Restrict to hapmap3 — genotype file will be much smaller later on.</li>
<li>Can in some cases by convenient to split the genotypes into the appropriate samples (discovery, if you are doing your own GWAS. Training/validation. Analysis sample), or to make files containing the sample IDs of the individuals that belong to each of these datasets.</li>
<li>Usually you will have the genotypes in bgen or plink format. In plink you will have three files: fam/bim/bed — but what’s in there? Example of the genotype files we will use in the exercise</li>
</ul>
<div id="bed-plink-format" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> bed – PLINK format<a href="background.html#bed-plink-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="background.html#cb1-1" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">+</span> <span class="dv">2</span></span>
<span id="cb1-2"><a href="background.html#cb1-2" tabindex="-1"></a>d</span>
<span id="cb1-3"><a href="background.html#cb1-3" tabindex="-1"></a><span class="do">## [1] 4</span></span></code></pre></div>
<!--
::: {style="background-color: #dedede; color: red; padding: 0.5em;"}
Some warning lalala
Hello **world**!
:::
-->
</div>
<div id="bgen-oxford-format" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> bgen – Oxford format<a href="background.html#bgen-oxford-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
</div>
<div id="summary-statistics" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Summary statistics<a href="background.html#summary-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Aligning summary stats with the genotypes: Flipping and ambigious SNPs. Describe and illustrate the problem with this.</p>
</div>
<div id="datasets" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Datasets<a href="background.html#datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This means we rely on several datasets (or splits of the same dataset):</p>
<ul>
<li><strong>Discovery sample</strong> You only need this if you do your own GWAS, but even if you use a ready-made set of summary statistics, you must make sure that the sample they used does not overlap with your analysis sample!</li>
<li><strong>Validation sample</strong> Call it a training/validation sample. Use it to evaluate your choices of hyperparameters etc.</li>
<li><strong>Reference sample</strong> Sometimes necessary, used to calculate SNP correlations (LD structure). Can overlap with validation/discovery sample, but generally not with analysis sample.</li>
<li><strong>Analysis sample</strong> The sample in which you want to construct a polygenic index for each individual.</li>
</ul>
<p>Requirements:</p>
<ul>
<li>We want the discovery sample to be large!</li>
<li>The reference/training sample can overlap.</li>
<li>The analysis sample should not overlap with discovery/reference/training samples!</li>
</ul>
</div>
<div id="polygenic-score-methods" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Polygenic score methods<a href="background.html#polygenic-score-methods" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="naive-method" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Naive method<a href="background.html#naive-method" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Idea: Weigh all SNPs together using the effect estimates from a GWAS. Say we have a simple GWAS of <span class="math inline">\(J\)</span> univariate regressions in the discovery sample <span class="math inline">\(\mathcal{D}\)</span>:
<span class="math display">\[y_i = \beta_0 + \beta_{j} \text{SNP}_{ij} + \varepsilon_{ij}, \quad i \in \mathcal{D}\]</span>
which gives us summary statistics <span class="math inline">\({\{}\hat{\beta_j}, se(\hat{\beta_j}){\}}_{j=1}^J\)</span>. We could construct the polygenic index as:
<span class="math display">\[\hat{PGI}_i = \sum_{j=1}^J \hat{\beta}_j \text{SNP}_{ij}, \quad i \in \mathcal{A}\]</span>
where we use the summary statistics from the discovery sample to weigh the SNPs for the individuals in the analysis sample <span class="math inline">\(\mathcal{A}\)</span>.</p>
<p>Issues:</p>
<ul>
<li><span class="math inline">\(\hat{\beta}_j\)</span> can be noisy though, how can we improve? What about incorporating the standard error <span class="math inline">\(se(\hat{\beta_j})\)</span> and selecting SNPs based on their signficance?</li>
<li>SNPs are correlated (LD, linkage disequilibrium), especially at shorter ranges, but the GWAS is a univariate regression, so we do not control for neighbouring SNPs that are correlated with SNP <span class="math inline">\(j\)</span>. Can we account for this somehow? Either in our selection of SNPs, or by adjusting the weights?</li>
</ul>
</div>
<div id="pruningclumping-and-threshold" class="section level3 hasAnchor" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Pruning/clumping and threshold<a href="background.html#pruningclumping-and-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Idea: Improve on the naive method by getting rid of SNPs that are highly correlated (in LD with eachother) and keep only SNPs that pass a test criterion.</p>
<p>Two approaches: Pruning and clumping.</p>
<ul>
<li>Pruning: We start at snp <span class="math inline">\(j\)</span>, and test correlations at <span class="math inline">\(j+1, j+2, \ldots, j+M\)</span> if the correlation exceeds a threshold then <span class="math inline">\(j\)</span> is dropped.</li>
<li>Clumping: We start at snp <span class="math inline">\(j\)</span>, and test correlations at <span class="math inline">\(j+1, j+2, \ldots, j+M\)</span>, if the correlation exceeds a threshold then the SNP with the highest p-value is dropped.</li>
</ul>
<!--  % description of clumping algo: https://rdrr.io/github/MRCIEU/ieugwasr/man/ld_clump.html
  % clumping vs pruning: https://www.biostars.org/p/343818/
  % clumping vs pruning: https://privefl.github.io/bigsnpr/articles/pruning-vs-clumping.html
  % clumping vs pruning, paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6904799/-->
<p>Clumping/thresholding requires us to select the p-value threshold, the squared correlation threshold, and the window size. In PRSICE r2 is 0.1 with a window of 250kb. PRSICE only searches over p-val threshold, not the other two hyperparameters.</p>
<p>We can prune (i.e. remove) SNPs that are highly correlated by looking at correlation in a reference sample. We can implement select SNPs via hard thresholding using a p-value cut off. How do we select the threshold? Grid search. Let’s work through the recipe.</p>
</div>
<div id="ldpred" class="section level3 hasAnchor" number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> LDPred<a href="background.html#ldpred" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Idea: Shrink effect estimates according to LD. Do bayesian selection on SNPs. Model:
<span class="math display">\[ yyyy \]</span></p>
<p>Two flavors: Infinitesimal (<span class="math inline">\(p = 1.0\)</span>) and non-infinitesimal model (<span class="math inline">\(p \in (0; 1)\)</span>)..</p>
<ul>
<li>The inf model reduces to a Gaussian prior on beta. Nice, this means it is fast as we can analytically derive the mean of the posterior:</li>
<li>The non-model is equivalent to a spike-slab model, so no analytical solution. We will have to sample the density. Slow!</li>
</ul>
<p>How do we select <span class="math inline">\(p\)</span>? Grid search, or a hyperprior and estimate it along with adjusted effect sizes. A lot of papers sticks to LDPred-Inf. To incorporate LD information we need the pairwise correlations between SNPs. We can call this correlation matrix the LD matrix, it is a triangular matrix, and, if we disregard long-range LD, a banded matrix. Nice! LDPred does not use LD blocks, but uses a sliding window around the current SNP when constructing this LD matrix.</p>
<p>LDPred1 is largely similar, but written in Python. LDPred2 is significantly faster and easier to work with from R, as it is not a commandline tool (if you can, then why not avoid platform dependent shell scripting).</p>
</div>
<div id="lassosum" class="section level3 hasAnchor" number="2.4.4">
<h3><span class="header-section-number">2.4.4</span> LASSOSUM<a href="background.html#lassosum" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Idea: …</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="in-practice.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
